app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyPresenter.java

```diff
@@ -185,18 +191,27 @@ private Completable showPendingTransaction(Payment transaction) {
     Log.d(TAG, "present: " + transaction);
     switch (transaction.getStatus()) {
       case COMPLETED:
-        return Completable.fromAction(view::showTransactionCompleted)
-            .andThen(Completable.timer(1, TimeUnit.SECONDS))
-            .observeOn(Schedulers.io())
-            .andThen(inAppPurchaseInteractor.getCompletedPurchase(transaction.getPackageName(),
+        return Completable.defer(() -> {
+          if (isBds) {
```

Comment:
is it possible to make this more abstract? the interactor should be the one that know that there is multiple types of payments